include/translated-post.php

```diff
@@ -410,4 +410,95 @@ protected function get_db_infos() {
 			'default_alias' => $GLOBALS['wpdb']->posts,
 		);
 	}
+
+	/**
+	 * Wraps `wp_insert_post` with language feature.
+	 *
+	 * @since 3.7
+	 *
+	 * @param array        $postarr {
+	 *     Optional. An array of elements that make up a post to update or insert.
+	 *     All `wp_insert_post` parameters plus the following.
+	 *
+	 *     @type string[] $translations The translation group to assign to the post with language slug as keys and post ID as values.
+	 * }
+	 * @param PLL_Language $language         The post language.
+	 * @param bool         $wp_error         Optional. Whether to return a WP_Error on failure. Default false.
+	 * @param bool         $fire_after_hooks Optional. Whether to fire the after insert hooks. Default true.
+	 * @return int|WP_Error The post ID on success. The value 0 or `WP_Error` on failure.
+	 */
+	public function insert( array $postarr, PLL_Language $language, bool $wp_error = false, bool $fire_after_hooks = true ) {
+		$post = wp_insert_post( $postarr, $wp_error, $fire_after_hooks );
+
+		if ( empty( $post ) || is_wp_error( $post ) ) {
+			// Something went wrong!
+			return $post;
+		}
+
+		$this->set_language( $post, $language );
+
+		if ( ! empty( $postarr['translations'] ) ) {
+			$this->save_translations(
+				$post,
+				array_merge(
+					$postarr['translations'],
+					array(
+						$language->slug => $post,
+					)
+				)
+			);
```

Comment:
This can be simplified as the merge is already don by `save_translations()`. See https://github.com/polylang/polylang/blob/3.6.4/include/translated-object.php#L487. I noticed that we can simplify for terms too.
```suggestion
			$this->save_translations( $post, $postarr['translations'] );
```